tests/Unit/Evolution/Systems/NewtonianEuler/Subcell/Test_ResizeAndComputePrimitives.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "Helpers/DataStructures/MakeWithRandomValues.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/PolytropicFluid.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+
+namespace {
+template <size_t Dim>
+void test(const gsl::not_null<std::mt19937*> gen,
+          const gsl::not_null<std::uniform_real_distribution<>*> dist,
+          const evolution::dg::subcell::ActiveGrid active_grid) {
+  using MassDensityCons = NewtonianEuler::Tags::MassDensityCons;
+  using EnergyDensity = NewtonianEuler::Tags::EnergyDensity;
+  using MomentumDensity = NewtonianEuler::Tags::MomentumDensity<Dim>;
+
+  using MassDensity = NewtonianEuler::Tags::MassDensity<DataVector>;
+  using Velocity = NewtonianEuler::Tags::Velocity<DataVector, Dim>;
+  using SpecificInternalEnergy =
+      NewtonianEuler::Tags::SpecificInternalEnergy<DataVector>;
+  using Pressure = NewtonianEuler::Tags::Pressure<DataVector>;
+
+  using cons_tags = tmpl::list<MassDensityCons, MomentumDensity, EnergyDensity>;
+  using ConsVars = Variables<cons_tags>;
+  using prim_tags =
+      tmpl::list<MassDensity, Velocity, SpecificInternalEnergy, Pressure>;
+  using PrimVars = Variables<prim_tags>;
+
+  const Mesh<Dim> dg_mesh{5, Spectral::Basis::Legendre,
+                          Spectral::Quadrature::GaussLobatto};
+  const Mesh<Dim> subcell_mesh = evolution::dg::subcell::fd::mesh(dg_mesh);
+
+  auto cons_vars = make_with_random_values<ConsVars>(
+      gen, dist,
+      active_grid == evolution::dg::subcell::ActiveGrid::Dg
+          ? dg_mesh.number_of_grid_points()
+          : subcell_mesh.number_of_grid_points());
+  PrimVars prim_vars{};
+  if (active_grid == evolution::dg::subcell::ActiveGrid::Subcell) {
+    prim_vars.initialize(subcell_mesh.number_of_grid_points(), 1.0);
+  }
+
+  std::unique_ptr<EquationsOfState::EquationOfState<false, 1>> eos =
+      std::make_unique<EquationsOfState::PolytropicFluid<false>>(1.4,
+                                                                 5.0 / 3.0);
+
+  auto box = db::create<db::AddSimpleTags<
+      evolution::dg::subcell::Tags::ActiveGrid, ::Tags::Variables<cons_tags>,
+      ::Tags::Variables<prim_tags>, ::domain::Tags::Mesh<Dim>,
+      evolution::dg::subcell::Tags::Mesh<Dim>,
+      hydro::Tags::EquationOfState<
+          std::unique_ptr<EquationsOfState::EquationOfState<false, 1>>>>>(
+      active_grid, cons_vars, prim_vars, dg_mesh, subcell_mesh, std::move(eos));
+
+  db::mutate_apply<NewtonianEuler::subcell::ResizeAndComputePrims<Dim>>(
+      make_not_null(&box));
+
+  REQUIRE(db::get<::Tags::Variables<prim_tags>>(box).number_of_grid_points() ==
+          cons_vars.number_of_grid_points());
```

Comment:
Why is this a `REQUIRE` vs a `CHECK`?

Reply:
because if this isn't true the check below is done on different grids and so is guaranteed to fail

Reply:
... but you want it to fail, so why is that bad? :P